library/smoothstreaming/src/main/java/com/google/android/exoplayer2/source/smoothstreaming/SsMediaSource.java

```diff
@@ -192,9 +192,7 @@ private SsMediaSource(SsManifest manifest, Uri manifestUri,
       AdaptiveMediaSourceEventListener eventListener) {
     Assertions.checkState(manifest == null || !manifest.isLive);
     this.manifest = manifest;
-    this.manifestUri = manifestUri == null ? null
-        : Util.toLowerInvariant(manifestUri.getLastPathSegment()).equals("manifest") ? manifestUri
-            : Uri.withAppendedPath(manifestUri, "Manifest");
+    this.manifestUri = manifestUri == null ? null : lastPathSegment.contains("manifest") && lastPathSegment.contains("(") && lastPathSegment.contains(")")) ? manifestUri : lastPathSegment.equals("manifest") ? manifestUri : Uri.withAppendedPath(manifestUri, "Manifest");
```

Comment:
- Does this actually compile? What is `lastPathSegment`?
- I'm not convinced about the logic. It matches all kinds of random things, like "xy)(manifest". Why not use `Util.toLowerInvariant(manifestUri.getLastPathSegment()).matches("XXX")` where "XXX" is a regular expression that more exactly describes what we consider a match?

Reply:
You are right - apologies for that, very sloppy of me.

And a reg-exp would be a much more elegant solution - will update